Description: The original Makefile either links everything dynamically or
 statically. This patch modifies the Makefile to link libjq statically and
 libonig2 dynamically.
Author: Simon Elsbrock
Forwarded: not-needed
Last-Update: 2015-10-11
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: jq/Makefile.am
===================================================================
--- jq.orig/Makefile.am	2015-10-11 03:50:10.822586743 +0200
+++ jq/Makefile.am	2015-10-11 12:08:38.703822442 +0200
@@ -40,12 +40,12 @@
 
 ### libjq
 
-lib_LTLIBRARIES = libjq.la
-libjq_la_SOURCES = ${LIBJQ_SRC}
-libjq_la_LIBADD = -lm
-libjq_la_LDFLAGS = -export-symbols-regex '^j[qv]_' -version-info 1:4:0
+#lib_LTLIBRARIES = libjq.la
+#libjq_la_SOURCES = ${LIBJQ_SRC}
+#libjq_la_LIBADD = -lm
+#libjq_la_LDFLAGS = -export-symbols-regex '^j[qv]_' -version-info 1:4:0
 
-include_HEADERS = jv.h jq.h
+#include_HEADERS = jv.h jq.h
 
 ### Running tests under Valgrind
 
@@ -64,8 +64,8 @@
 ### Error injection for testing
 
 if ENABLE_ERROR_INJECTION
-lib_LTLIBRARIES += libinject_errors.la
-libinject_errors_la_SOURCES = inject_errors.c
+lib_LTLIBRARIES = libinject_errors.la
+libinject_errors_la_SOURCES = inject_errors.c ${LIBJQ_SRC}
 libinject_errors_la_LIBADD = -ldl
 libinject_errors_la_LDFLAGS = -module
 endif
@@ -83,13 +83,14 @@
 main.c: version.h
 
 bin_PROGRAMS = jq
-jq_SOURCES = main.c version.h
-jq_LDFLAGS = -static-libtool-libs
-jq_LDADD = libjq.la -lm
-
-if ENABLE_ALL_STATIC
-jq_LDFLAGS += -all-static
-endif
+jq_SOURCES = main.c version.h ${LIBJQ_SRC}
+#jq_LDFLAGS = -static-libtool-libs
+jq_CFLAGS = $(AM_CFLAGS)
+jq_LDADD = -lm -lonig
+
+#if ENABLE_ALL_STATIC
+#jq_LDFLAGS += -all-static
+#endif
 
 ### Tests (make check)
 
